### ISSUE TITLE
Keybinds stop working when using Caps-lock

### ISSUE THREAD

**Issue by lavandongen** [pqrs-org/Karabiner-Elements#2114]
I have a simple keybind to replace the ± button with the ~ button. However it seems that in many cases where I need to enable the Caps-lock key the application crashes and causes my Apple Magic Keyboard 2 to disconnect. Forcing me to turn it off and on again, only to find my keybinds are no longer working.

I end up having to restart my entire laptop for the problem to go away.

I've encountered this issue many, many times over the ast few months since I have installed Catalina,  am not loving this and would love to see a fix quickly.

**Comment by gaoyuqiang** [comment]
I have the same problem !  I  replace caps-lock to esc , but it often dones't work  , and I must  restart my macbook! Now I  restart my macbook  two or three times one day ! I am confused!

**Comment by cristiangauma** [comment]
I have the latest Macbook, and I'm facing this problem since I have it (with latest Catalina 10.15.2). I only have Karabiner installed to override the "weird" space character that has Mac (cmd + space). But since I installed it, the keyboard stops working randomly when I use Caps-lock, using either the mac keyboard or an external keyboard.

When it happens, all the keys stops working for the internal keyboard or external keyboard (ESC, FN, touchbar and touch id work), the weird thing is that if I was using a no-Mac external keyboard, if I connect a Mac keyboard once the internal and no-Mac keyboard are blocked, then the external Mac keyboard works while the others are blocked. Also, when I use the virtual keyboard and unlock de Caps-lock, the light of the internal/external Caps-lock goes away, but the keyboard keeps frozen.

The only fix that I found is to reboot the laptop, I haven't been able to find another way to let the keyboard work again.

**Comment by lavandongen** [comment]
@tekezo The issue still persist in 12.9.0 unfortunately

**Comment by brianshim** [comment]
Same problem here.  Sometimes after pressing caps lock, keyboard becomes unresponsive. Everything else still works.  2019 MacBook Pro 16" running Catalina.

**Comment by tekezo** [comment]
I need more information to investigate this issue since I cannot reproduce it.

Is turn off Manipulate LED checkbox for your all connected keyboards solve the issue?

<img width="1000" alt="Screen Shot 2020-01-21 at 11 14 20" src="https://user-images.githubusercontent.com/659178/72769789-41a28e80-3c3f-11ea-855f-35c85f2a8ff3.png">

**Comment by lavandongen** [comment]
I have now turned off all my LEDs and will notify here if the issue persists.

**Comment by lavandongen** [comment]
@tekezo I have not ran into any issues since turning off LED manipulation. It seems that that was causing an issue somewhere! Hope this helps.

**Comment by LukaszKrulikowski** [comment]
I have the same issue. During 2 months it happened 3 times. I checked that I also had LED manipulation ON. I turned it off. We will see if this will solve the problem.

**Comment by lavandongen** [comment]
@tekezo I just had the Apple Magic Keyboard stop working again, even without the LED manipulation on. I had been working for nearly the entire day and needed Caps Lock for something. I type at a moderate tempo, but as soon as I hit the Caps Lock while writing a line of code the keyboard stopped working and I had to turn it on and off again. After which the keybindings no longer worked.

**Comment by lavandongen** [comment]
I have just experienced the issue again

**Comment by queenkjuul** [comment]
I am having this issue with 12.9.0 on Catalina 10.15.2, using a late-2015 iMac and a "standard" Apple keyboard (aluminum chiclet, A1243). Changing the "Manipulate LED" setting does not affect the issue, it happens either way. My keyboard will entirely stop responding, if I unplug it and plug it back in the keyboard will start working but Karabiner will not until the system is rebooted.

**Comment by j-mroz** [comment]
Have the same problem.

### Steps to reproduce: 
Press caps lock repeatedly. 

### Actual result
 Keyboard stops working

### Expected result
Keyboardf should be still working

# System info
MacOS Catalina 10.15.3 (19D76) on macBook pro 15" 2018

**Comment by neodoubleo** [comment]
I'm also having this issue on a 2015 iMac, a 2016 Macbook Pro, and a 2018 Macbook Pro. At this point, I'm fine if somebody could at least figure out how to restart karabiner properly, because like you guys said, even if I restart the external keyboard, key bindings no longer work.

**Comment by teddywing** [comment]
Related to #2035 and #2038

**Comment by b-onc** [comment]
Having the same issue. Sometimes using caps-lock freezes my entire keyboard, restart is the only solution.

**Comment by kkarczmarczyk** [comment]
I have the same issue! sometimes (but not always), after presing caps-lock, keyboard stops working (or repeat last pressed key indefinitely, for example if I open notepad and press "a" and later i press caps-lock I can see "aaaaaaaaaaaaaaaaaaa" and keyboard freezes) - only restart (not sleep, full restart needed) helps...

**Comment by snapwich** [comment]
Same issue here. My computer actually crashed and upon re-login I think some update was performed (by OSX or my enterprise domain admin). None of my simple or complex modifications work anymore, including capslock override and some key remappings.

**Comment by onelaview** [comment]
Found this problem as well.

In my case, I bind caps-lock to operate as Hyper key and Esc key if pressed alone. Once happened, my external keyboard just stops working entirely. Need to re-plug it in order to have it works again. But then caps-lock binding as Hyper or Esc key just stops working. Restart Karabiner-Elements doesn't help unfortunately. Only restart Mac will solve it.

![image](https://user-images.githubusercontent.com/512696/81673196-726c3500-9475-11ea-8c5e-c2b67866a2b3.png)

I'm on Catalina 10.15.4 on MacBook Pro (15-inch, 2016) with Karabiner-Elements 12.9.0

**Comment by lavandongen** [comment]
I am currently working on configuring some 404 pages for a client. To test that I often just spam the keyboard, but because I strike the keyboard too fast Karabiner seems to flip out more than usual and I need to restart the entire laptop once or twice a day.

I am still running the latest version of both MacOs and Karabiner. I wish this issue, which has been known for quite a while now, could be resolved soon!

**Comment by eync** [comment]
I am having aswell issues with pressing caps-lock which freeze's my laptop's keyboard and changes my mouse's behaviour. External Keyboard starts to behave so differently. Been experiencing with different solutions but none of them seems to work. Only solution is to restart reboot laptop. 
I'm on Catalina 10.15.5 Macbook Pro (2018 13inch) and Karabiner-Elements version 12.9.0.

**Comment by bstone100** [comment]
I have had this issue occur every-so-often the entire time I have been on Catalina... the only solution still being a restart. Hope someone can find a solution.

**Comment by ivan-leschinsky** [comment]
I had the same issue and already forgot about Karabiner (used few times with different profiles). I uninstalled Karabiner and all symptoms fixed

**Comment by ionstormx** [comment]
This might just be coincidental, but I haven't faced any issues since i've turned off "Manipulate LED".
- I'm on the latest version of macOS
- I do use the Caps Lock as a hyper key

**Comment by embedded-creations** [comment]
> This might just be coincidental, but I haven't faced any issues since i've turned off "Manipulate LED".
> 
> * I'm on the latest version of macOS
> * I do use the Caps Lock as a hyper key

Same here, it’s been weeks since I made the change, and the issues have stopped completely

**Comment by joshave443** [comment]
I have the LED Manipulation box unchecked and this still regularly happens.

macOS 15.5.5
WASD Code-39 Keyboard
<img width="921" alt="Screen Shot 2020-06-25 at 3 42 09 PM" src="https://user-images.githubusercontent.com/62728988/85802758-72a95280-b6fa-11ea-82d3-c8857351a7c3.png">

**Comment by neodoubleo** [comment]
Hey all. 12.10.0 is released today. Please post if you still run into the issue after the update.
-oo

**Comment by Inflatable-Boat** [comment]
I have been having the the exact same issue, just happened today again unfortunately. Sometimes when hitting caps-lock (more often when using other function keys) the keyboard freezes - putting my Mac to sleep makes the keyboard work again, but keybinds will be ignored, the only solution is a full restart of my Mac.

**Comment by domq** [comment]
> Hey all. 12.10.0 is released today. Please post if you still run into the issue after the update.
> -oo

Updated today, crashed 25 minutes later ☹

**Comment by radaczynski** [comment]
Is this a duplicate of https://github.com/pqrs-org/Karabiner-Elements/issues/2035?

**Comment by teddywing** [comment]
> Is this a duplicate of #2035?

It appears to be the same issue, as far as I can tell.

**Comment by grebulon** [comment]
This looks like a duplicate of [#2247](https://github.com/pqrs-org/Karabiner-Elements/issues/2247)

**Comment by helgridly** [comment]
+1, also seeing this issue. I added a simple modification to rebind caps lock to VK_NONE (thus disabling it); this at least stops me accidentally breaking my keyboard (at the cost of not being able to yell on the internet). Turning off Manipulate LED also worked for me.

**Comment by pbriggs28** [comment]
I am having the same issue. Sometimes it happens up to twice a week, sometimes it doesn't happen for three weeks.

I too feel like typing speed might be involved, it seems to happen when I am typing fast. 
I want to say this only happens going from non-caps to caps-lock mode, but need to verify this.

When this happens both my external keyboard (Dell WK717, connected via BT) and internal keyboard (2017 15-inch MacBook Pro) stop working and my internal keyboard shows the caps lock key lit (will check if my external shows lit next time). If I turn my external keyboard off/on it starts working again but Karabiner no longer works (acts as though I have no mappings). I don't see anything 

Nothing seems to fix the issue besides restarting:
- Restarting Karabiner (Misc > Restart Karabiner-Elements)
- Quitting Karabiner (Misc > Quit Karabiner-Elements) and restarting
- Force quitting through Activity Monitor, all

MacBook: 2017 15-inch MacBook Pro
OS: Catalina 10.15.6 (19G73)
Karabiner: 12.10.0
External keyboard: Dell WK717, connected via BT

I did have "Manipulate LED" enabled on some devices so I have removed that and will see if it fixes it.

What data points do you want us to gather to help you troubleshoot? Maybe if there was some logging you could add around this (I know, easier said than done) we could post logs when this happens. Here's what I will gather whenever this happens from now on, let me know if there is anything else you want:

- Typing fast (y/n)
- Caps lock transition (to caps lock/from caps lock)
- Manipulate LED enabled
- Caps lock LED lit on external keyboard
- Caps lock LED lit on internal keyboard
- Logs

**Comment by brianafm** [comment]
Sigh same issue. All I wanted was for my Magic Keyboard’s home key to work like a home key is supposed to work. I uninstalled Karabiner (maybe not all the way because I still see it in my Keyboard settings) but this really sucks. I can’t keep restarting the computer, especially inconvenient during a Zoom or conference call

**Comment by ionstormx** [comment]
My initial findings - https://github.com/pqrs-org/Karabiner-Elements/issues/2114#issuecomment-644659610

Today I noticed another contributor to the keyboard stalling. After weeks of non-issue, the keyboard stalled **when I turned on and used Airdrop**

**Comment by s-surineni** [comment]
I am facing the same issue. Any idea when this will be fixed?

**Comment by neodoubleo** [comment]
Ya, I don't think there's a plan for getting this fixed. We keep reporting the issue over an over again, without any resolution. I've actually started using [BetterTouchTool](https://folivora.ai/) instead. It lets me do what I've been doing with Karabiner perfectly (vim cursor keys). So depending on what your needs are, you guys might want to check it out as an alternative (and it has tons of other great features). -oo

**Comment by bartvdo** [comment]
I have the same issue. Hit capslock and the keyboard freezes. This is the main keyboard on the mac, not attached. trackpad still works. Need to restart to get the keyboard running.

I disabled the led, no change. This is based on a vanilla installation, no configuration has been done (using it to have the function keys working again after a keyboard switch).

**Comment by kerryj89** [comment]
Very strange bug.  I just experienced it myself.

**Comment by deborahkim** [comment]
Beta seems to have resolved the issue! Thanks @MuhammedZakir!

I'll report back if I experience this issue again, but I would recommend folks to try it. Definitely a macOS issue for me.

**Comment by jsrozner** [comment]
I started using a different karabiner default config:
caps lock = {escape when tapped, control when held}
control = {hyper when held, caps lock on/off when tapped}

I have not experienced any more keyboard freezes using this new config. 

https://github.com/jsrozner/dotfiles/tree/master/other_config/karabiner

**Comment by Bielik20** [comment]
I do not use caps lock intensively (out of feat at this point), but I feel like this latest updated helped. Few times my keyboard  froze for a few second after using caps lock but then it unfroze (after clicking other keys) which is great improvement as it doesn't require system restart. I will keep monitoring the situation.

**Comment by cristiancheles** [comment]
Did anyone use Karabiner lately? Does the bug still occur on the latest version or in the beta?

**Comment by Bielik20** [comment]
I haven't experienced it for some time now 👍 
I think we could close this issue and reopen should it happen again.

**Comment by bartvdo** [comment]
I haven't had the issue since going to the beta. I've changed back to the regular release cycle and the current version doesn't have this problem anymore.

**Comment by deborahkim** [comment]
Agreed, same here. No issues now. 👍

I also went to the beta at first, then switched to the latest version, and it seems like they folded in the fix successfully.

No issues upgrading to Big Sur, either.

**Comment by dav1d-wright** [comment]
I am on 13.4.0 and still have this issue. Will try installing the beta version.

**Comment by dav1d-wright** [comment]
Update: the issue is still there for me (OSX 11.3, Karabiner 13.4.4)

**Comment by jonathanpaulpano** [comment]
This is not an issue with Karabiner but it's the macOS itself. It started with Catalina and now Big Sur. I've been in macOS since OS X 10.6 up to now and never have I experience this kind of issue until Catalina.

Pressing the caps lock sometimes surges the USB port power out. You can check this when the keyboard locks out, then go to Apple System Report and check the USB port where the keyboard was in. You will see that there is no current available and current required.

Unplugging and replugging the USB in the same port where the keyboard was in do not solve the issue because the OS seems to remember that the keyboard was in that port before, therefore it will not give power to the USB port. But if you put another USB device that the keyboard was in, the OS will give power again.

Attaching USB keyboard to your extra ports will cause it to get working again but, if you press the caps lock button again the surge in that port will cause the keyboard to not work again. Now you have two ports that will not give the power to the keyboard and so on.

**Comment by dav1d-wright** [comment]
Thanks for the explanation @jonathanpaulpano. Would it be possible for Karabiner to write a workaround?

**Comment by dav1d-wright** [comment]
Bump

**Comment by GorvGoyl** [comment]
capslock key is still disabling the keyboard.
macOS: Monterey 12.2.1

**Comment by kerryj89** [comment]
Last time I experienced it was in 2020 https://github.com/pqrs-org/Karabiner-Elements/issues/2114#issuecomment-701168919

Today it said hello to me again.  I am using a wireless keyboard so not sure about it surging anything out for me.

**Comment by nejmeijer** [comment]
Unclear how this differs from #2247. I am tagging this as a duplicate for the time being.
